src/__experimental__/components/date/date-picker.component.js

```diff
@@ -76,28 +92,19 @@ const DatePicker = ({
     },
   };
 
-  const picker = (
-    <StyledDayPicker>
-      <DayPicker
-        {...datePickerProps}
-        containerProps={{ style: disablePortal ? {} : containerPosition }}
-        ref={datepicker}
-      />
-    </StyledDayPicker>
-  );
-
-  if (disablePortal) {
-    return picker;
-  }
-
   return (
-    <StyledPortal
-      onReposition={() =>
-        setContainerPosition(getContainerPosition(window, inputElement))
-      }
+    <Popover
+      placement="bottom-start"
+      reference={inputElement}
+      modifiers={popoverModifiers(size)}
```

Comment:
I think this will cause Popover instance to be destroyed and created on each render now.